### PR TITLE
(#1088) Add test to check interruption in wait

### PR DESCRIPTION
**Pull request by Umbrah**
This is for #1088 
I added test which verifies case when thread is interrupted while waiting.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1124](https://github.com/yegor256/cactoos/issues/1124) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1124](https://github.com/yegor256/cactoos/issues/1124) is assigned to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), here is [why](https://www.0crat.com/footprint/C63314D6Z/ae261bdc-7826-4f01-97d1-851238dd525c); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/func/RetryTest.java [hunk 1]
```diff
@@ -74,4 +73,29 @@ public void runsFuncConditionMultipleTimes() {
...
         ).affirm();
     }
+
+    @Test
+    public void processInterruptExceptionOnWait() {
+        final Thread main = Thread.currentThread();
+        new Thread(
+            () -> {
+                try {
+                    Thread.sleep(500);
```
Comment (ilyakharlamov) [inline src/test/java/org/cactoos/func/RetryTest.java]: Using Thread.sleep in tests is a bad idea. It not only slows down the test suite run time, but it also creates fragile tests that can pass or fail depending on the environment timing.

File: src/test/java/org/cactoos/func/RetryTest.java [hunk 2]
```diff
@@ -74,4 +73,29 @@ public void runsFuncConditionMultipleTimes() {
...
+                }
+                main.interrupt();
+            }).start();
+        new Assertion<>(
+            "Must be interrupted on wait",
+            () -> new Retry<>(
+                input -> {
+                    throw new IllegalArgumentException("Happens");
+                },
+                (attempts) -> false,
```
Comment (ilyakharlamov) [inline src/test/java/org/cactoos/func/RetryTest.java]: No need for the parentheses.

File: src/test/java/org/cactoos/func/RetryTest.java [hunk 3]
```diff
@@ -74,4 +73,29 @@ public void runsFuncConditionMultipleTimes() {
...
+                Duration.ofMillis(1000)
```
Comment (ilyakharlamov) [inline src/test/java/org/cactoos/func/RetryTest.java]: Implicit type conversion is not recommended - use 1000L

**Comment by Umbrah** [comment]
@ilyakharlamov I replaced Thread.sleep with checking for TIMED_WAITING thread state, and fixed 2 other comments. Please check

File: src/test/java/org/cactoos/func/RetryTest.java [hunk 4]
```diff
@@ -74,4 +73,29 @@ public void runsFuncConditionMultipleTimes() {
...
         ).affirm();
     }
+
+    @Test
+    public void processInterruptExceptionOnWait() {
+        final Thread main = Thread.currentThread();
+        new Thread(
+            () -> {
+                while (true) {
+                    if (Thread.State.TIMED_WAITING.equals(main.getState())) {
```
Comment (ilyakharlamov) [inline src/test/java/org/cactoos/func/RetryTest.java]: Enums are safe to compare using `==` instead of equals.

**Comment by Umbrah** [comment]
@ilyakharlamov Updated

**Comment by ilyakharlamov** [comment]
@llorllale looks OK to me

**Comment by Umbrah** [comment]
@llorllale ping

**Comment by Umbrah** [comment]
@llorllale ping

**Comment by Umbrah** [comment]
@llorllale ping

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
The job [#1124](https://github.com/yegor256/cactoos/issues/1124) is now out of scope

**Comment by 0crat** [comment]
Job [#1124](https://github.com/yegor256/cactoos/issues/1124) is not in the agenda of @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), can't inspect

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (21 days), architects (@llorllale) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Pull request merged**